wave_optimization/tests/factor_graph/factor_graph_test.cpp

```diff
@@ -0,0 +1,39 @@
+#include "wave/wave_test.hpp"
+#include "wave/optimization/factor_graph/factor_graph.hpp"
+
+
+namespace wave {
+
+TEST(FactorGraph, addUnaryFactor) {
+    FactorGraph graph;
+    graph.addUnaryFactor<PoseVar, double>(1, 2.0);
```

Comment:
Let's have a quick discussion during Monday's meeting on finalizing this interface.  Also, how will this interface connect to the Ceres solver, @chutsu ?